src/main/scala/nodes/images/Convolver.scala

```diff
@@ -24,24 +26,29 @@ class Convolver(
     imgChannels: Int,
     whitener: Option[ZCAWhitener] = None,
     normalizePatches: Boolean = true,
-    varConstant: Double = 10.0)
+    varConstant: Double = 10.0,
+    patchStride: Int = 1)
   extends Transformer[Image, Image] {
 
   val convSize = math.sqrt(filters.cols/imgChannels).toInt
   val convolutions = filters.t
 
-  val resWidth = imgWidth - convSize + 1
-  val resHeight = imgHeight - convSize + 1
+  val resWidth = (imgWidth - convSize + 1)
+  val resHeight = (imgHeight - convSize + 1)
 
   override def apply(in: RDD[Image]): RDD[Image] = {
     in.mapPartitions(Convolver.convolvePartitions(_, resWidth, resHeight, imgChannels, convSize,
-      normalizePatches, whitener, convolutions, varConstant))
+      normalizePatches, whitener, convolutions, varConstant, patchStride))
   }
 
   def apply(in: Image): Image = {
-    var patchMat = new DenseMatrix[Double](resWidth*resHeight, convSize*convSize*imgChannels)
+
+    val outWidth = math.ceil(resWidth/patchStride).toInt
```

Comment:
well, they're both `Int`, right now, so this is either a bug or not doing anything?